src/cascadia/TerminalApp/GlobalAppSettings.idl

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import "AppLogic.idl";
+import "ColorScheme.idl";
+import "AppKeybindings.idl";
+import "Command.idl";
+import "TerminalSettings.idl";
+
+namespace TerminalApp
+{
+    // MIDL 3 allows for structs to hold nullable types
+    // Though IReference is a WinRT object, MIDL 3
+    // handles all of the ownership logic for us.
+    // Docs: https://docs.microsoft.com/en-us/uwp/midl-3/intro#types
+    struct LaunchPosition
+    {
+        Windows.Foundation.IReference<Int64> X;
+        Windows.Foundation.IReference<Int64> Y;
+    };
+
+    [default_interface] runtimeclass GlobalAppSettings {
+        GlobalAppSettings();

```

Comment:
not necessary unless you want somebody **outside the library** to be allowed to say "Make a new blank Global Settings" (projected type)

`make`-ing Implementation types doesn't need the ctor in the idl or the factory.

Reply:
(Lumping your last comment with this one). CascadiaSettings needed it in its constructor for whatever reason.

Reply:
Nvmd